config/routes.rb

```diff
@@ -1,10 +1,11 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+   # Defines the root path route ("/")
+  # root "posts#index"
+  get "users/:user_id/posts", to: "posts#index", as: "user_posts"
+  get "users/:user_id/posts/:id", to: "posts#show", as: "user_post"
+  get "users", to: "users#index", as: "users"
+  get "users/:id", to: "users#show", as: "user"
```

Comment:
- [OPTIONAL] I think it would be better to use [resource routing](https://guides.rubyonrails.org/v3.2.2/routing.html#resource-routing-the-rails-default) for defining routes. It not only simplifies your code but also aligns with Rails conventions, making your application more maintainable and developer-friendly ✨⛏

```ruby
resources :users, only: [:index, :show] do
  resources :posts, only: [:index, :show]
end
```